File: Easy/Array/3194. Minimum Average of Smallest and Largest Elements/Solution.java
class Solution {
    public double minimumAverage(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);

        double min = Double.MAX_VALUE;

        for(int i = 0; i < n/2; i++){
            double avg = (nums[i] + nums[n - 1 - i]) / 2.0;
            min = Math.min(min, avg);
        }

        return min;
    }
}

File: Easy/Array/3194. Minimum Average of Smallest and Largest Elements/Solution.kt
class Solution {
    fun minimumAverage(nums: IntArray): Double {
        
        nums.sort()

        var minAverage = Double.MAX_VALUE

        var start = 0
        var end = nums.size - 1

        while(start < end){
            val avg = (nums[start] + nums[end]) / 2.0
            minAverage = minOf(minAverage, avg)
            
            start++
            end--
        }

        return minAverage
    }
}
